orc/src/main/java/com/netflix/iceberg/orc/ORC.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2018 Hortonworks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.iceberg.orc;
+
+import com.google.common.base.Preconditions;
+import com.netflix.iceberg.PartitionSpec;
+import com.netflix.iceberg.Schema;
+import com.netflix.iceberg.io.InputFile;
+import com.netflix.iceberg.io.OutputFile;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.orc.OrcFile;
+import org.apache.orc.Reader;
+import org.apache.orc.TypeDescription;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+public class ORC {
+  private ORC() {
+  }
+
+  public static WriteBuilder write(OutputFile file) {
+    return new WriteBuilder(file);
+  }
+
+  public static class WriteBuilder {
+    private final OutputFile file;
+    private Schema schema = null;
+    private PartitionSpec spec = null;
+    private Configuration conf = null;
+    private final Properties tableProperties = new Properties();
+    private Map<String, byte[]> metadata = new HashMap<>();
+
+    private WriteBuilder(OutputFile file) {
+      this.file = file;
+    }
+
+    public WriteBuilder partitionSpec(PartitionSpec spec) {
+      this.spec = spec;
+      return this;
+    }
+
+    public WriteBuilder metadata(String property, String value) {
+      metadata.put(property, value.getBytes(StandardCharsets.UTF_8));
+      return this;
+    }
+
+    public WriteBuilder tableProperties(Properties properties) {
```

Comment:
Is there a reason to use Properties to pass options? I'd normally use a `config(String, String)` method instead.

Reply:
Ok, what I was trying to do was:

* Not create a new Configuration. They are expensive from both time and memory.
* Not change the Configuration that was passed in, since I don't own it.

That said, I'm not against changing it so that I clone the passed in Configuration. I've had bad experiences making changes to a configuration that was passed in, so I'd rather avoid that.

Reply:
I'd prefer to hide the `Properties` so that the caller only has to work with the builder. Properties can be used to accumulate the configuration and passed in. If this is a good way to avoid needing to change the `Configuration`, then I'm all for it.